spacy/gold.pyx

```diff
@@ -494,9 +553,47 @@ def read_json_object(json_corpus_section):
     YIELDS (tuple): The reformatted data.
     """
     for json_doc in json_corpus_section:
-        tuple_doc = json_to_tuple(json_doc)
-        for tuple_paragraph in tuple_doc:
-            yield tuple_paragraph
+        yield from json_to_tuple(json_doc)
+
+
+class DocTuple(object):
+
```

Comment:
Unfortunately we have been working on a similar idea on `develop`: a new [`Example`](https://github.com/explosion/spaCy/blob/develop/spacy/gold.pyx#L711) class that keeps track of annotations as well as of the original doc/text. Here, we could potentially add the `doc_id`. You would then define it when you call `json_to_examples` (renamed from `json_to_tuple`) in a similar way as you're doing it now.